File: forensics-btmgen/src/main/java/de/burger/forensics/plugin/engine/SourceFileGuards.java
package de.burger.forensics.plugin.engine;

import java.io.File;
import java.util.function.Consumer;

public final class SourceFileGuards {
    private SourceFileGuards() {}

    public static boolean shouldSkipLargeFile(File file, long maxBytes, Consumer<String> debug) {
        if (maxBytes <= 0) return false;
        long length = file.length();
        if (length > maxBytes) {
            if (debug != null) {
                debug.accept("Skipping large file (" + length + " bytes > limit " + maxBytes + "): " + file.getAbsolutePath());
            }
            return true;
        }
        return false;
    }
}

File: forensics-btmgen/src/main/kotlin/de/burger/forensics/plugin/engine/SourceFileGuards.kt
package de.burger.forensics.plugin.engine

import java.io.File

fun shouldSkipLargeFile(file: File, maxBytes: Long, debug: (String) -> Unit): Boolean {
    if (maxBytes <= 0) return false
    val length = file.length()
    if (length > maxBytes) {
        debug("Skipping large file (${length} bytes > limit $maxBytes): ${file.absolutePath}")
        return true
    }
    return false
}
